extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.css

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * .amp-lightboxed gets added to the element when displayed in the lightbox.
+ * .amp-lightboxed is designed to provide a reasonable default behaviour for
+ * lightboxing of arbitary elements.
+ * Both components and page authors are free to customize the style of the
+ * element when inside the lightbox.
+ * Example:
+ *    .myDiv {
+ *       color: blue;
+ *     }
+ *
+ *     .myDiv.amp-lightboxed {
+ *       color: red;
+ *     }
+ */
+.amp-lightboxed {
+  position: fixed !important;
+  z-index: 2147483645 !important;
+  width: 100%;
+  box-sizing: border-box;
+  margin: 0;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+}
+
+/**
+ * The following rule is needed to reset the stacking context.
+ * By resetting the properties that create new stacking context on the
+ * ancestors of the item being lightboxed, we end up with a single stacking
+ * context and therefore the z-index will be absolute. See https://goo.gl/uqY5CN
+ * Code is copied from Webkit's FullScreen Css https://goo.gl/gjbKoI
+ */
+.-amp-lightboxed-ancestor {
+  z-index: auto !important;
+  position: static !important;
```

Comment:
This one concerns me a bit and I'm not entirely understanding it. This will preclude us from ever allowing opacity of the lightbox to be less than 1. If it's our only choice - that's probably ok. But I was wondering if something like this could also work:

```
.-amp-lightboxed-ancestor, .amp-lightboxed {
  z-index: MAX_VALUE;
}
.-amp-lightboxed-ancestor > :not(.amp-lightboxed):not(.-amp-lightboxed-ancestor) {
  opacity: 0.99999;
}
```

This solution, instead of stacking the lightbox chain to the root, would instead unstack all other siblings.